CMakeLists.txt

```diff
@@ -52,6 +52,13 @@ set(F3D_SYSTEM_PROCESSOR "${F3D_SYSTEM_PROCESSOR}-bits")
 option(F3D_EXCLUDE_DEPRECATED "Exclude deprecated functions and options" OFF)
 mark_as_advanced(F3D_EXCLUDE_DEPRECATED)
 
+option(F3D_ENABLE_CXX17_API "Enable libf3d C++17 API" ON)
```

Comment:
I don't think it is necessary.  
C++17 is needed to build F3D anyway.  
The issue is variant-related APIs not compatible with C++11 for libf3d consumers.  
In this case, a simple preprocessor variable to hide the API is enough.  
Something like:
```cpp
#ifndef F3D_DISABLE_CXX17_API
variant related stuff
#endif
```
This variable can be enabled with CMake **by the consumer** or by doing something like:
```cpp
#define F3D_DISABLE_CXX17_API
#include <f3d/options.h>
```

Let me know if you need to talk about that offline

Reply:
Indeed, thatd be much simpler.

However I do need a CMake option to build the SDK tests without C++17 in order to check it actually works as expected imo.

Reply:
Just add `F3D_DISABLE_CXX17_API` definition to the CMakr target when building the SDK tests?

Reply:
Sometimes we need it, sometimes we dont, ill try to find the best way

Reply:
Then a #define inside the cxx file that needs C++11 only would work